pkg/controller/deploymentv1beta1/deployment_util.go

```diff
@@ -0,0 +1,1006 @@
+/*
```

Comment:
these are really really really ugly. Tons of duplicated code. Moreover, this has nothing to do with a controller anymore and hence should not be under pkg/controller.

Reply:
With kubectl moving out of main kube repo, what is the plan with these? @fabianofranz @pwittrock

Reply:
@sttts Thanks for the feedback. This PR is WIP and not intended for review without getting rid of the duplicated codes. I am still finding a way to de-duplicate the util codes. I will mark the PR as WIP, and inform you when it is ready for review.

Reply:
oh, ok. Just saw it being assigned to me :)

Reply:
Sorry for not marking the PR WIP beforehand. Thanks for the feedback! :)